File: _includes/code/python/local.quickstart.query.rag.py
# RAG
import weaviate

client = weaviate.connect_to_local()

questions = client.collections.get("Question")

# highlight-start
response = questions.generate.near_text(
    query="biology",
    limit=2,
    grouped_task="Write a tweet with emojis about these facts."
)
# highlight-end

print(response.generated)  # Inspect the generated text

client.close()  # Free up resources
# END RAG

File: _includes/code/typescript/local.quickstart.query.rag.ts
// RAG
import weaviate, { WeaviateClient } from 'weaviate-client';

const client: WeaviateClient = await weaviate.connectToLocal();

// highlight-start
const questions = client.collections.get('Question');

const result = await questions.generate.nearText(
  'biology',
  {
    groupedTask: 'Write a tweet with emojis about these facts.',
  },
  {
    limit: 2,
  }
);
// highlight-end

console.log(result.generated);

client.close(); // Close the client connection
// END RAG
